plugin/audit_log/buffer.c

```diff
@@ -189,6 +189,7 @@ int audit_log_buffer_write(audit_log_buffer_t *log, const char *buf, size_t len)
       log->write_func(log->write_func_data, buf, len, LOG_RECORD_COMPLETE);
       audit_log_buffer_resume(log);
     }
+    my_atomic_add64(&audit_log_buffer_size_overflow, (int64)1);
```

Comment:
Redundant cast:

```suggestion
    my_atomic_add64(&audit_log_buffer_size_overflow, 1);
```

Reply:
> Redundant cast:
> 
> Suggested change Beta 
> [Give us feedback](https://www.research.net/r/SuggestedChanges)
>     my_atomic_add64(&audit_log_buffer_size_overflow, (int64)1);
>     my_atomic_add64(&audit_log_buffer_size_overflow, 1);

@laurynas-biveinis if I don't cast, it gives me an error:

```
/home/marcelo/git/percona-server/plugin/audit_log/buffer.c: In function ‘audit_log_buffer_write’:
/home/marcelo/git/percona-server/plugin/audit_log/buffer.c:193:54: error: incompatible type for argument 2 of ‘my_atomic_add64’
     my_atomic_add64(&audit_log_buffer_size_overflow, 1);
                                                      ^
In file included from /home/marcelo/git/percona-server/plugin/audit_log/buffer.c:22:0:
/home/marcelo/git/percona-server/include/my_atomic.h:168:24: note: expected ‘U_64 {aka union <anonymous>}’ but argument is of type ‘int’
 static inline int ## S my_atomic_add ## S(                      \
                        ^
/home/marcelo/git/percona-server/include/my_atomic.h:211:1: note: in expansion of macro ‘make_atomic_add’
 make_atomic_add(64)
 ^~~~~~~~~~~~~~~
``` 

For 5.7 it works fine, only 5.6 that requires the implicit casting.